### PR TITLE
Added support for some movement differences on Bedrock <-> Java (Draft)

### PR DESCRIPTION
**Pull request by Oryxel**
- Bedrock doesn't seems to have modifyVelocityZero (0.003/0.005D), the movement appear to just slowly going down until it reaches 0 (or some extremely small value).
- If you moving on both X and Z axis your movement input get multiplied by 1f / Math.sqrt(2), you can get this value by debugging using ProxyPass and check for input value bedrock send.

Also this is tested on [here](https://github.com/Oryxel/Boar), an anti-cheat that I'm currently working on that is incomplete but is good enough to check for (1e-4) movement differences if you just walking around normally.

### CODE REVIEW

File: src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/minecraft/entity/MixinLivingEntity.java [hunk 1]
```diff
@@ -75,10 +76,23 @@ public abstract class MixinLivingEntity extends Entity {
...
+
+    @Shadow public float forwardSpeed;
+
     public MixinLivingEntity(EntityType<?> type, World world) {
         super(type, world);
     }
 
+    @Inject(method = "tickMovement", at = @At(value = "INVOKE", target = "Lnet/minecraft/util/profiler/Profiler;push(Ljava/lang/String;)V", ordinal = 3, shift = At.Shift.AFTER))
+    private void bedrockXZInput(CallbackInfo ci) {
+        if (this.sidewaysSpeed != 0 && this.forwardSpeed != 0) {
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/minecraft/entity/MixinLivingEntity.java:89]: Would need a check for bedrock version

File: src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/minecraft/entity/MixinLivingEntity.java [hunk 2]
```diff
@@ -242,7 +256,9 @@ private void removeShieldSlotPreference(ItemStack stack, CallbackInfoReturnable<
 
     @ModifyConstant(method = "tickMovement", constant = @Constant(doubleValue = 0.003D))
     private double modifyVelocityZero(final double constant) {
-        if (ProtocolTranslator.getTargetVersion().olderThanOrEqualTo(ProtocolVersion.v1_8)) {
+        if (ProtocolTranslator.getTargetVersion().equals(BedrockProtocolVersion.bedrockLatest)) {
+            return 0D;
+        }else if (ProtocolTranslator.getTargetVersion().olderThanOrEqualTo(ProtocolVersion.v1_8)) {
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/minecraft/entity/MixinLivingEntity.java:261]:
```suggestion
        } else if (ProtocolTranslator.getTargetVersion().olderThanOrEqualTo(ProtocolVersion.v1_8)) {
```

File: src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/minecraft/entity/MixinLivingEntity.java [hunk 3]
```diff
@@ -75,10 +76,23 @@ public abstract class MixinLivingEntity extends Entity {
...
+    @Shadow public float forwardSpeed;
+
     public MixinLivingEntity(EntityType<?> type, World world) {
         super(type, world);
     }
 
+    @Inject(method = "tickMovement", at = @At(value = "INVOKE", target = "Lnet/minecraft/util/profiler/Profiler;push(Ljava/lang/String;)V", ordinal = 3, shift = At.Shift.AFTER))
+    private void bedrockXZInput(CallbackInfo ci) {
+        if (this.sidewaysSpeed != 0 && this.forwardSpeed != 0) {
+            float value = (float) (1F / Math.sqrt(2));
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/minecraft/entity/MixinLivingEntity.java:90]: The value could be a static marked with Unique and prefixed with viaFabricPlus$

**Review by FlorianMichael — CHANGES_REQUESTED** [review]
Beside the styling/general things; RK and I both don't have the time to review/verify this anytime soon and it's also questionable if this should wait for ViaBedrock to get more features/be more usable.

**Comment by Oryxel** [comment]
Done, I will leave this here until ViaBedrock to get is more usable.

**Pull request closed without merging**